lib/containers/utils.pm

```diff
@@ -315,4 +315,96 @@ sub reset_container_network_if_needed {
     }
 }
 
+sub test_volumes {
+    my %args = @_;
+    my $runtime = $args{runtime};
+
+    # From https://docs.docker.com/storage/bind-mounts/
+    # The --mount flag does not support z or Z options for modifying selinux labels.
+    my $Z = $runtime eq "podman" ? ",Z" : "";
+
+    my $test_dir = "test_dir";
+    my $test_file = "test_file";
+    my $test_image = "test_image";
+    my $test_container = "test_container";
+    my $test_volume = "test_volume";
+
+    # Create a directory with the sticky-bit set because of rootless podman
+    assert_script_run("mkdir -m 1777 $test_dir");
+
+    # Create Dockerfile with VOLUME defined
+    assert_script_run("echo -e 'FROM busybox\nVOLUME /$test_dir' > $test_dir/Dockerfile");
+
+    # Build image
+    assert_script_run("$runtime build -t $test_image -f $test_dir/Dockerfile $test_dir/");
+
+    assert_script_run("touch $test_dir/$test_file");
+
+    # Test --volumes-from
+    assert_script_run("$runtime run -d --name $test_container $test_image");
+    assert_script_run("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir | grep $test_file");
```

Comment:
```suggestion
    validate_script_output("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir", sub { $_ !~ m/$test_file/ };);
```

Just a suggestion, but `validate_script_output` is easier to debug IMHO.

Reply:
> Just a suggestion, but `validate_script_output` is easier to debug IMHO.

Shouldn't it be:

    validate_script_output("$runtime run --rm --volumes-from $test_container $test_image ls /$test_dir", sub { $_ ~ m/^$test_file$/ };); ?